src/NeuralNetwork/index.js

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2019 ml5
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+
+/*
+Generic NeuralNetwork class
+*/
+
+import * as tf from '@tensorflow/tfjs';
+
+class NeuralNetwork {
+  /**
+   * Create a Neural Network.
+   * @param {object} options - An object with options.
+   */
+  constructor(options) {
+    // TODO: create the model based on many more options and defaults
+
+    const inputUnits = options.input || 2;
+    const hiddenUnits = Math.floor(inputUnits / 2) + 1;
+    const outputUnits = options.output || 1;
+
+    this.model = tf.sequential();
+    const hidden = tf.layers.dense({
+      units: hiddenUnits,
+      inputShape: [inputUnits],
+      activation: 'sigmoid',
+    });
+    const output = tf.layers.dense({
+      units: outputUnits,
+      activation: 'sigmoid',
+    });
+    this.model.add(hidden);
+    this.model.add(output);
+
+    const LEARNING_RATE = 0.25;
+    const optimizer = tf.train.sgd(LEARNING_RATE);
+
+    this.model.compile({
+      optimizer,
+      loss: 'meanSquaredError',
+      metrics: ['accuracy'],
+    });
+
+    this.training_xs = [];
+    this.training_ys = [];
+  }
+
+  addData(xs, ys) {
```

Comment:
Not sure what the name of this function should be or whether it's ml5's job really to compile the data. However, this I view as analogous to `addExample()` with KNN  or featureExtractor transfer learning. It's possible that we'll want to have a `ml5.data` feature for the library. At the moment I am considering "batches" to be beyond the scope of ml5 and handled internally.